# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/debugger/Debugger.cpp
# Commit: e802face8e04
# Full Hash: e802face8e042d03933c1495d36fd02a9def7dd9
# Author: Brian Hackett <bhackett1024@gmail.com>
# Date: 2019-07-25 21:51:57
# Regressor Bug: 1554524
# File Overlap Count: 1
# Description:
#   Bug 1554524 Part 4 - Add Debugger interface for instrumenting scripts, r=jimb,tcampbell.
# ==============================================================================

diff -r fc204daba6f3 -r e802face8e04 js/src/debugger/Debugger.cpp
--- a/js/src/debugger/Debugger.cpp	Mon Jul 08 09:39:52 2019 -1000
+++ b/js/src/debugger/Debugger.cpp	Mon Jul 08 09:41:21 2019 -1000
@@ -48,6 +48,7 @@
 #include "vm/AsyncFunction.h"
 #include "vm/AsyncIteration.h"
 #include "vm/GeckoProfiler.h"
+#include "vm/Instrumentation.h"
 #include "vm/JSContext.h"
 #include "vm/JSObject.h"
 #include "vm/Realm.h"
@@ -6408,6 +6409,19 @@
   return wrapVariantReferent(cx, referent);
 }
 
+bool DebugAPI::getScriptInstrumentationId(JSContext* cx,
+                                          HandleObject dbgObject,
+                                          HandleScript script,
+                                          MutableHandleValue rval) {
+  Debugger* dbg = Debugger::fromJSObject(dbgObject);
+  DebuggerScript* dbgScript = dbg->wrapScript(cx, script);
+  if (!dbgScript) {
+    return false;
+  }
+  rval.set(dbgScript->getInstrumentationId());
+  return true;
+}
+
 static bool DebuggerSource_construct(JSContext* cx, unsigned argc, Value* vp) {
   JS_ReportErrorNumberASCII(cx, GetErrorMessage, nullptr, JSMSG_NO_CONSTRUCTOR,
                             "Debugger.Source");